src/librustc_typeck/check/mod.rs

```diff
@@ -3445,6 +3469,51 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 
+    /// Given a vec of evaluated `FullfillmentError`s and an `fn` call expression, we walk the
+    /// `PathSegment`s and resolve their type parameters to see if any of the `FullfillmentError`s
+    /// were caused by them. If they were, we point at the corresponding type argument's span
+    /// instead of the `fn` call path span.
+    fn point_at_type_arg_instead_of_call_if_possible(
```

Comment:
@Centril you're gonna love this 😬

Reply:
Jesus...
12 braces is just obscene. 😅
But at least it's in a new method!